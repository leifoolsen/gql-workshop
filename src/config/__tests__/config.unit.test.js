const importFresh = require('import-fresh');

describe('Config', () => {
  describe('development', () => {
    let env;

    beforeAll(() => {
      env = process.env.NODE_ENV;
      process.env.NODE_ENV = 'development';
    });

    afterAll(() => {
      process.env.NODE_ENV = env;
    });

    test('NODE_ENV="development', () => {
      expect(process.env.NODE_ENV).toEqual('development');
    });

    test('skal returnere konfigurasjon for NODE_ENV="development', () => {
      expect(process.env.NODE_ENV).toEqual('development');
      const config = importFresh('../index.js');
      expect(config).toMatchSnapshot();
    });
  });

  describe('test', () => {
    let env;

    beforeAll(() => {
      env = process.env.NODE_ENV;
      process.env.NODE_ENV = 'test';
    });

    afterAll(() => {
      process.env.NODE_ENV = env;
    });

    test('NODE_ENV="test', () => {
      expect(process.env.NODE_ENV).toEqual('test');
    });

    test('skal returnere konfigurasjon for NODE_ENV="test', () => {
      const config = importFresh('../index.js');
      expect(config).toMatchSnapshot();
    });
  });

  describe('production', () => {
    let env;

    beforeAll(() => {
      env = process.env.NODE_ENV;
      process.env.NODE_ENV = 'production';
    });

    afterAll(() => {
      process.env.NODE_ENV = env;
    });

    test('NODE_ENV="production', () => {
      expect(process.env.NODE_ENV).toEqual('production');
    });

    test('skal returnere konfigurasjon for NODE_ENV="production', () => {
      const config = importFresh('../index.js');
      expect(config).toMatchSnapshot();
    });
  });
});
