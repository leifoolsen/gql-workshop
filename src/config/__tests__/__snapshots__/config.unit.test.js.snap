// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config development skal returnere konfigurasjon for NODE_ENV="development 1`] = `
Object {
  "default": Object {
    "apiServer": Object {
      "apiPath": "/graphql",
      "compression": false,
      "host": "localhost",
      "port": 3001,
      "scheme": "http",
    },
    "isDev": true,
    "isHot": false,
    "isProd": false,
    "isTest": false,
    "logger": Object {
      "console": Object {
        "level": "debug",
        "silent": false,
      },
      "file": Object {
        "filename": "./logs/application.log",
        "level": "error",
        "silent": true,
      },
    },
    "server": Object {
      "apiPath": "/graphql",
      "compression": false,
      "contentBase": "client",
      "historyApiFallback": Object {
        "verbose": true,
      },
      "host": "localhost",
      "port": 3000,
      "proxy": Object {
        "/graphql": Object {
          "changeOrigin": true,
          "logLevel": "debug",
          "secure": false,
          "target": "http://localhost:3001",
        },
      },
      "publicPath": "/",
      "scheme": "http",
    },
    "useProxy": false,
  },
}
`;

exports[`Config production skal returnere konfigurasjon for NODE_ENV="production 1`] = `
Object {
  "default": Object {
    "apiServer": Object {
      "apiPath": "/graphql",
      "compression": false,
      "host": "localhost",
      "port": 3001,
      "scheme": "http",
    },
    "isDev": true,
    "isHot": false,
    "isProd": false,
    "isTest": false,
    "logger": Object {
      "console": Object {
        "level": "debug",
        "silent": false,
      },
      "file": Object {
        "filename": "./logs/application.log",
        "level": "error",
        "silent": true,
      },
    },
    "server": Object {
      "apiPath": "/graphql",
      "compression": false,
      "contentBase": "client",
      "historyApiFallback": Object {
        "verbose": true,
      },
      "host": "localhost",
      "port": 3000,
      "proxy": Object {
        "/graphql": Object {
          "changeOrigin": true,
          "logLevel": "debug",
          "secure": false,
          "target": "http://localhost:3001",
        },
      },
      "publicPath": "/",
      "scheme": "http",
    },
    "useProxy": false,
  },
}
`;

exports[`Config test skal returnere konfigurasjon for NODE_ENV="test 1`] = `
Object {
  "default": Object {
    "apiServer": Object {
      "apiPath": "/graphql",
      "compression": false,
      "host": "localhost",
      "port": 3001,
      "scheme": "http",
    },
    "isDev": true,
    "isHot": false,
    "isProd": false,
    "isTest": false,
    "logger": Object {
      "console": Object {
        "level": "debug",
        "silent": false,
      },
      "file": Object {
        "filename": "./logs/application.log",
        "level": "error",
        "silent": true,
      },
    },
    "server": Object {
      "apiPath": "/graphql",
      "compression": false,
      "contentBase": "client",
      "historyApiFallback": Object {
        "verbose": true,
      },
      "host": "localhost",
      "port": 3000,
      "proxy": Object {
        "/graphql": Object {
          "changeOrigin": true,
          "logLevel": "debug",
          "secure": false,
          "target": "http://localhost:3001",
        },
      },
      "publicPath": "/",
      "scheme": "http",
    },
    "useProxy": false,
  },
}
`;
